romanNumeralsExercise/romanNumeralsExercise.js

```diff
@@ -0,0 +1,143 @@
+const inputNumber = prompt("Enter any digit between one and 3000: ");
+
+const seperateInputNumbers = [];
+
+let zeros;
+
+// loop through each value in input, round off each value to nearest 1000,100 or 10
+function numberRoundOff() {   
+    if (inputNumber.length == 4) {
+        zeros = '000';
+        for (let i = 0; i<inputNumber.length; i++) {
+        const numberA = inputNumber[i] + zeros ;
+        zeros = zeros.slice(0,-1);
+        seperateInputNumbers.push(Number(numberA));
+        }
+    } else if (inputNumber.length == 3) {
+        zeros = '00';
+        for (let i = 0; i<inputNumber.length; i++) {
+        const numberA = inputNumber[i] + zeros ;
+        zeros = zeros.slice(0,-1);
+        seperateInputNumbers.push(Number(numberA));
+        }
+    } else if (inputNumber.length == 2) {
+        zeros = '0';
+        for (let i = 0; i<inputNumber.length; i++) {
+        const numberA = inputNumber[i] + zeros ;
+        zeros = zeros.slice(0,-1);
+        seperateInputNumbers.push(Number(numberA));
+        }
+    } else if (inputNumber.length == 1) {
+        seperateInputNumbers.push(Number(inputNumber));
+    } else if ( inputNumber.length > 4) {
+        console.log("The number you have inserted is too large")
+    }
+    else {
+        console.log('You have not inserted a number');
+    }
+    return seperateInputNumbers;
+}
+numberRoundOff();
+console.log('Input numbers seperated and rounded off: ')
+console.log(seperateInputNumbers);
+
+
+let romanNumeralValue = '';
+let testValue;
+let baseNum;
+
+function romanNumerals() {
+    for (let i=0; i<seperateInputNumbers.length; i++){
```

Comment:
You can also look at using [forEach()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) in the future for loops.

Reply:
One thing I'd suggest is trying to refactor this solution so that you have the Roman numerals mapped to numbers. I think that would be easier to read :)

Reply:
So as you loop over your array, you'd check in the mapping if the number corresponds, and if so, append the roman numeral to your string.